pkg/cmd/issue/edit/edit.go

```diff
@@ -141,75 +150,125 @@ func editRun(opts *EditOptions) error {
-	editable.Labels.Default = issue.Labels.Names()
-	editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
-	projectItems := map[string]string{}
-	for _, n := range issue.ProjectItems.Nodes {
-		projectItems[n.Project.ID] = n.ID
-	}
-	editable.Projects.ProjectItems = projectItems
-	if issue.Milestone != nil {
-		editable.Milestone.Default = issue.Milestone.Title
-	}
-
-	if opts.Interactive {
-		err = opts.FieldsToEditSurvey(&editable)
-		if err != nil {
-			return err
-		}
-	}
-
+	// Fetch editable shared fields once for all issues.
 	apiClient := api.NewClientFromHTTP(httpClient)
-	opts.IO.StartProgressIndicator()
+	opts.IO.StartProgressIndicatorWithLabel("Fetching repository information")
 	err = opts.FetchOptions(apiClient, repo, &editable)
 	opts.IO.StopProgressIndicator()
 	if err != nil {
 		return err
 	}
 
-	if opts.Interactive {
-		editorCommand, err := opts.DetermineEditor()
-		if err != nil {
-			return err
-		}
-		err = opts.EditFieldsSurvey(&editable, editorCommand)
-		if err != nil {
-			return err
-		}
+	// Update all issues in parallel.
+	if opts.workers == 0 {
+		opts.workers = 10
 	}
+	g, ctx := errgroup.WithContext(context.Background())
+	g.SetLimit(opts.workers)
+
+	issueURLs := make([]string, 0, len(issues))
+	issuesChan := make(chan string)
+	doneCh := make(chan any)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				doneCh <- nil
+				return
+			case issueURL, ok := <-issuesChan:
+				if ok {
+					issueURLs = append(issueURLs, issueURL)
+				}
+			}
+		}
+	}()
+
+	opts.IO.StartProgressIndicatorWithLabel(fmt.Sprintf("Updating %d issues", len(issues)))
+	for _, issue := range issues {
+		// Copy variables to capture in the go routine below.
+		issue := *issue
+		editable := editable
+
+		g.Go(func() error {
+			editable.Title.Default = issue.Title
+			editable.Body.Default = issue.Body
+			editable.Assignees.Default = issue.Assignees.Logins()
+			editable.Labels.Default = issue.Labels.Names()
+			editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
+			projectItems := map[string]string{}
+			for _, n := range issue.ProjectItems.Nodes {
+				projectItems[n.Project.ID] = n.ID
+			}
+			editable.Projects.ProjectItems = projectItems
+			if issue.Milestone != nil {
+				editable.Milestone.Default = issue.Milestone.Title
+			}
 
-	opts.IO.StartProgressIndicator()
-	err = prShared.UpdateIssue(httpClient, repo, issue.ID, issue.IsPullRequest(), editable)
+			// Allow interactive prompts for one issue; failed earlier if multiple issues specified.
+			if opts.Interactive {
+				editorCommand, err := opts.DetermineEditor()
+				if err != nil {
+					return err
+				}
+				err = opts.EditFieldsSurvey(&editable, editorCommand)
+				if err != nil {
+					return err
+				}
+			}
+
+			err = prShared.UpdateIssue(httpClient, repo, issue.ID, issue.IsPullRequest(), editable)
```

Comment:
What are your thoughts on having the Go routine only contain this function call and moving all the manipulation of `editable` struct and prompting code outside of it? Seems to me that if we can reduce the amount of code that gets executed inside the Go routine to only the operations we need to be async we will reduce potential bugs that could occur.

Reply:
The `Editable` will need to be copied, and if we do the interface prompt outside it means bifurcating the code for a single issue vs. multiple. I think it'd make the code more confusing, personally. That's why I added the comment to explain, though, that we could only be interactive with a single issue so there should be no fear of trying to prompt under different go routines.

Reply:
I am not sure I am following the reasoning here. Perhaps I didn't explain my idea well enough though. Was just thinking moving line 205 `g.Go(func() error {` down to line 231. I tried it out locally and it seems to work fine without any additional changes.

Reply:
One added benefit of this is we could allow interactive editing with multiple issues, something that we might want to support in the future.

Reply:
I get it now. Thanks for clarifying, and I agree that's a nice benefit if you wanted to support it in the future. Obviously the total execution time is gated on user input, but there'd be no reason not to do the update of the previous issue while prompting for subsequent issues.

In fact, should I just add support for that now. When @mislav and I were discussing this idea in #5782 I don't think we considered that - probably just assumed there'd be a contention issue once parallelized, but with your idea it wouldn't be. Just need to be clear which issue we're prompting for which either 1) changes the UX slightly for a single issue or, 2) we just don't do for a single issue.

Reply:
I think we should hold off on allowing interactivity for the moment since that was not what was initially agreed upon and I would like to run the idea by the team first. I think it would be fairly easy to iterate and add in the feature in a follow up PR if we decide we want it.

Reply:
I opened #7285 to track. I'd be happy to implement it when we complete this PR. I'm working on all the changes discussed herein and will apply the same for PRs.